source/adapters/opencl/usm.cpp

```diff
@@ -275,14 +276,14 @@ UR_APIEXPORT ur_result_t UR_APICALL urEnqueueUSMFill(
       cl_ext::MemBlockingFreeName, &USMFree));
 
   cl_int ClErr = CL_SUCCESS;
-  auto HostBuffer = static_cast<uint64_t *>(
+  auto HostBuffer = static_cast<unsigned char *>(
       HostMemAlloc(CLContext, nullptr, size, 0, &ClErr));
   CL_RETURN_ON_FAILURE(ClErr);
 
-  auto NumValues = size / sizeof(uint64_t);
-  auto NumChunks = patternSize / sizeof(uint64_t);
-  for (size_t i = 0; i < NumValues; i++) {
-    HostBuffer[i] = static_cast<const uint64_t *>(pPattern)[i % NumChunks];
+  auto NumChunks = size / patternSize;
+  for (size_t i = 0; i < NumChunks; i++) {
+    auto Dest = HostBuffer + i * patternSize;
+    memcpy(Dest, pPattern, patternSize);
   }
```

Comment:
Maybe we don't need to divide `size` by `patternSize`?

```suggestion
  auto *End = HostBuffer + size;
  for (auto *Iter = HostBuffer; Iter < End; Iter += patternSize) {
    std::memcpy(Iter, pPattern, patternSize);
  }
```

Note this is completely untested so I could be totally wrong.

Reply:
That's much better, thanks!